src/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -367,6 +369,7 @@ def _save_topk_checkpoint(self, trainer: "pl.Trainer", monitor_candidates: Dict[
             self._save_none_monitor_checkpoint(trainer, monitor_candidates)
 
     def _save_checkpoint(self, trainer: "pl.Trainer", filepath: str) -> None:
+        self._last_checkpoint_saved = filepath
```

Comment:
Is there a specific reason why you set this attribute before `save_checkpoint` and not after with `last_global_step_saved`?

Reply:
Yeah, we could save it after since it seems we don't need it as part of the state.